pennylane/devices/default_qubit_jax.py

```diff
@@ -212,6 +212,15 @@ def sample_basis_states(self, number_of_states, state_probability):
             key = jax.random.PRNGKey(np.random.randint(0, 2**31))
         else:
             key = self._prng_key
+        if jnp.ndim(state_probability) == 2:
+            # TODO [dwierichs]: Check whether the following introduces strong correlations between
+            # different samples along the broadcasted axis
+            return jnp.array(
+                [
+                    jax.random.choice(key, number_of_states, shape=(shots,), p=prob)
+                    for prob in state_probability
+                ]
+            )
```

Comment:
This is analogue to `DefaultQubit`. However, as mentioned in the todo above, I'm not sure reusing the key like this is okay?

Reply:
Good question 🤔

Reply:
It did, I fixed this by creating new keys for each prob in `state_probability` via `jax.random.split` :)